services/localzone.c

```diff
@@ -1407,17 +1407,70 @@ local_data_answer(struct local_zone* z, struct module_env* env,
 		lz_type == local_zone_inform_redirect) &&
 		qinfo->qtype != LDNS_RR_TYPE_CNAME &&
 		lr->rrset->rk.type == htons(LDNS_RR_TYPE_CNAME)) {
+		uint8_t* ctarget;
+		size_t ctargetlen = 0;
+
 		qinfo->local_alias =
 			regional_alloc_zero(temp, sizeof(struct local_rrset));
 		if(!qinfo->local_alias)
 			return 0; /* out of memory */
-		qinfo->local_alias->rrset =
-			regional_alloc_init(temp, lr->rrset, sizeof(*lr->rrset));
+		qinfo->local_alias->rrset = regional_alloc_init(
+			temp, lr->rrset, sizeof(*lr->rrset));
 		if(!qinfo->local_alias->rrset)
 			return 0; /* out of memory */
-		/* TODO local_alias->rrset change cnam etarget */
 		qinfo->local_alias->rrset->rk.dname = qinfo->qname;
 		qinfo->local_alias->rrset->rk.dname_len = qinfo->qname_len;
+		get_cname_target(lr->rrset, &ctarget, &ctargetlen);
+		if(!ctargetlen)
+			return 0; /* invalid cname */
+		if(dname_is_wild(ctarget)) {
+			/* synthesize cname target */
+			struct packed_rrset_data* d;
+			uint8_t newtarget[LDNS_MAX_DOMAINLEN];
```

Comment:
Other buffers are `LDNS_MAX_DOMAINLEN+1`. Also for rpz.c:646 `wc[LDNS_MAX_DOMAINLEN]`, the +1 is elsewhere but not here.

Reply:
The newtarget[] is actually not used at all and removed in a later commit (https://github.com/NLnetLabs/unbound/pull/73/commits/344f12dd992286b36fc0679d50389782b33d6308)

Reply:
added +1 for the wc[] in rpz.c